gcc_deployment_script.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+
+set -e
+
+usage() {
+  cat <<EOF
+  usage: $0
+  mandatory options:  -a ARTIFACTS_DIR
+                      -n REPOSITORY_NAME
+                      -g GIT_REF
+                      -u REMOTE_USER
+                      -i REMOTE_IP
+                      -t REMOTE_DEST_DIR
+                      [-r]
+  This script creates an appropriately named directory
+  with artifacts and deploys it to a provided deployment
+  server.
+  If -r is present, release version of deployment is performed,
+  otherwise build version.
+EOF
+}
+
+send_dir() {
+  scp_src="$isolate_dir/."
+  remote_host="$remote_user@$remote_ip"
+  remote_dest="$remote_dest_dir"
+  scp_dest="$remote_host:$remote_dest"
+
+  echo -e "\n- Sending dir to deployment server"
+  echo -e "  src: $scp_src"
+  echo -e "  dest: $scp_dest"
+  scp -r "$scp_src" "$scp_dest"
+}
+
+rename_dir() {
+  deployment_dir_name=""
+  if [[ -z "$release_version" ]]; then
+    deployment_dir_name="$repository_name""_build_""$git_ref"
+  else
+    deployment_dir_name="$repository_name""_""$git_ref"
+  fi
+
+  temp_deployment_parent_dir="$(dirname "$temp_deployment_dir")"
+  deployment_dir="$temp_deployment_parent_dir/$deployment_dir_name"
+
+  echo -e "\n- Renaming deployment directory"
+  echo -e "  old: $temp_deployment_dir"
+  echo -e "  new: $deployment_dir"
+  mv "$temp_deployment_dir" "$deployment_dir"
+}
+
+create_dir_with_artifacts() {
+  isolate_dir="$(mktemp -d)"
+  temp_deployment_dir="$isolate_dir/$repository_name/$deployment_version/deployment_artifacts"
+
+  mkdir -p "$temp_deployment_dir"
+
+  echo -e "\n- Copying artifacts"
+  echo -e "  src:$artifacts_dir/*"
+  echo -e "  dest:$temp_deployment_dir"
+  cp -r "$artifacts_dir/." "$temp_deployment_dir"
```

Comment:
I don't know why you would need the /. characters here.

Reply:
Because I would like to copy what's inside the directory, not the whole directory.